File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.kt
class Solution {
    fun solution(card1: Array<String>, card2: Array<String>, goal: Array<String>): String {
        val c1 = card1.toMutableList()
        val c2 = card2.toMutableList()
        for (g in goal) {
            if (c1.size > 0 && c1[0] == g) {
                c1.removeAt(0)
            }
            else if (c2.size > 0 && c2[0] == g) {
                c2.removeAt(0)
            }
            else {
                return "No"
            }
        }
        return "Yes"
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
from collections import deque
def solution(c1, c2, goal):
    flag = True
    q1 = deque(c1)
    q2 = deque(c2)
    g = deque(goal)
    while g:
        t = g.popleft()
        if q1:
            e1 = q1.popleft()
        if q2:
            e2 = q2.popleft()
        if t == e1: q2.appendleft(e2)
        elif t == e2: q1.appendleft(e1)
        else:
            flag = False
            break
    return "Yes" if flag else "No"